datasets/diff_helper.py

```diff
@@ -0,0 +1,118 @@
+import requests
+import os
+import json
+import sys
+
+DATASET_FIELDS = [
+  'name',
+  'workflow_key',
+  'visibility'
+]
+
+CATALOGUE_FIELDS = [
+  'description',
+  'creator',
+  'contactPoint',
+  'publisher',
+  'license',
+  'versionInfo',
+  'keyword',
+  'identifier',
+  'rights',
+]
+
+DICTIONARY_FIELDS = [
+  'name',
+  'code',
+  'description',
+  'fields',
+  'lookups'
+]
+
+class DiffHelper:
+  def toplevel_diff(previous, current):
+    diff = {}
+    for field in DATASET_FIELDS:
+      if (field in current and previous[field] != current[field]): diff[field] = current[field]
+    return diff
+    
+  def catalogue_diff(previous, current):
+    diff = {}
+    for field in CATALOGUE_FIELDS:
+      if (field in current and previous[field] != current[field]): diff[field] = current[field]
+    return diff
+
+  def dictionaries_diff(previous, current):
+    diff = {}
+    codes, dictionaries = DiffHelper.new_and_deleted_dictionaries(previous, current)
+    dictionaries += DiffHelper.changed_dictionaries(current, previous, codes)
+    return dictionaries
+
+  def dataset_diff(original, data):
+    diff = {}
+
+    if DiffHelper.is_equal(original, data): return diff
+    
+    diff = DiffHelper.toplevel_diff(original, data)
+    catalogueUpdates = DiffHelper.catalogue_diff(original['catalogue'], data['catalogue'])
+    dictionaryUpdates = DiffHelper.dictionaries_diff(original['dictionaries'], data['dictionaries'])
+    if catalogueUpdates: diff['catalogue'] = catalogueUpdates
+    if dictionaryUpdates: diff['dictionaries'] = dictionaryUpdates
+
+    return diff
+
+  def sanitize_dict(original):
+    modified = original.copy()
+    for key in original:
+      if key not in DICTIONARY_FIELDS or original[key] == None:
+        modified.pop(key)
+    return modified
+
+  def is_equal(original, data):
```

Comment:
I don't think we need this method as it's a wrapper around the standard equality check.

Reply:
Yeah not quite sure why this was wrapped!